velox/common/base/BitUtil.h

```diff
@@ -355,9 +356,17 @@ void forEachBit(
         if (!word) {
           return;
         }
-        while (word) {
-          func(idx * 64 + __builtin_ctzll(word));
-          word &= word - 1;
+        if (kAllSet == word) {
+          const size_t start = idx * 64;
+          const size_t end = (idx + 1) * 64;
+          for (size_t row = start; row < end; ++row) {
+            func(row);
```

Comment:
In a microbenchmark where `func` is small and therefore doesn't affect inlining, I would definitely expect to see a win from this change. However now `func` is invoked twice, which could cause code size to increase and affect inlining decisions. Have you run other function benchmarks to check the impact of this change?

Reply:
Thats a great point ; Looking at clang documentation(https://clang.llvm.org/docs/analyzer/developer-docs/IPA.html#deciding-when-to-inline)  though It looks like function length isnt the primary factor that determines inlining but more other factors (variadic functions, fn definition not avilable etc..). I would reckon in most cases it should inline, however I will try and run some tests with larger functions to determine this.

Reply:
Discussed offline: other benchmarks in the codebase are not negatively impacted, so we're fine here.